bam/reader.go

```diff
@@ -355,9 +355,17 @@ func parseAux(aux []byte) []sam.Aux {
 type buffer struct {
 	off  int
 	data []byte
+	err  error
 }
 
 func (b *buffer) bytes(n int) []byte {
+	if b.err != nil {
+		return nil
+	}
+	if b.len() < n {
+		b.err = io.ErrUnexpectedEOF
+		return nil
+	}
 	s := b.off
```

Comment:
wouldn't it be better to check that `b.len() < n + b.off` ?

Reply:
`b.len()` is the number of remaining bytes in the buffer (see [here](https://github.com/biogo/hts/blob/d0dadd4feeafd9428b7186d376fa658ab1ac075d/bam/reader.go#L374)).

Reply:
ah. right.